File: protobuf_mqtt_bridge/examples/python/mqtt_listener.py
import paho.mqtt.client
import autoware_protobuf.std_msgs.msg.string_pb2
import autoware_protobuf.std_msgs.msg.uint32_pb2

def create_mqtt_client():
    client = paho.mqtt.client.Client()
    client.on_connect = on_connect
    client.message_callback_add("/string",on_string)
    client.message_callback_add("/uint32",on_uint32)
    client.on_message = on_message
    client.connect("localhost", 1883, 60)
    return client

def on_connect(client, userdata, flags, result):
    print(f"Connected: {result}")

def on_message(client, userdata, msg):
    print("unknown topic:", msg.topic)

def on_string(client, userdata, msg):
    message = autoware_protobuf.std_msgs.msg.string_pb2.String()
    message.ParseFromString(msg.payload)
    print(message.data)

def on_uint32(client, userdata, msg):
    message = autoware_protobuf.std_msgs.msg.uint32_pb2.UInt32()
    message.ParseFromString(msg.payload)
    print(message.data)

if __name__ == "__main__":
    client = create_mqtt_client()
    client.subscribe("/string")
    client.subscribe("/uint32")
    client.loop_forever()

File: protobuf_mqtt_bridge/examples/node/mqtt_listener.js
const uint32 = require('./autoware_protobuf/uint32');
const mqtt = require('mqtt')
const client  = mqtt.connect('mqtt://localhost')

protobuf.load("proto/autoware_protobuf/std_msgs/msg/uint32.proto", function(err, root) {
    if (err) { throw err; }
})

client.on('connect', function () {
    client.subscribe('/uint32')
})

client.on('message', function (topic, payload) {
    console.log(payload)
    client.end()
})
